### PR TITLE
r/aws_s3_bucket: Add S3 Same-Region Replication acceptance test

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Add an acceptance test for the `aws_s3_bucket` resource to verify that [S3 Same-Region Replication](https://aws.amazon.com/about-aws/whats-new/2019/09/amazon-s3-introduces-same-region-replication/) just works.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
n/a
```

Output from acceptance testing:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSS3Bucket_SameRegionReplicationSchemaV2'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -parallel 20 -run=TestAccAWSS3Bucket_SameRegionReplicationSchemaV2 -timeout 120m
=== RUN   TestAccAWSS3Bucket_SameRegionReplicationSchemaV2
=== PAUSE TestAccAWSS3Bucket_SameRegionReplicationSchemaV2
=== CONT  TestAccAWSS3Bucket_SameRegionReplicationSchemaV2
--- PASS: TestAccAWSS3Bucket_SameRegionReplicationSchemaV2 (69.94s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	69.970s
```

### CODE REVIEW

File: aws/resource_aws_s3_bucket_test.go [hunk 1]
```diff
@@ -1862,6 +1862,66 @@ func TestAccAWSS3Bucket_ReplicationSchemaV2(t *testing.T) {
...
 }
 
+func TestAccAWSS3Bucket_SameRegionReplicationSchemaV2(t *testing.T) {
+	rInt := acctest.RandInt()
+	region := testAccGetRegion()
+	partition := testAccGetPartition()
+	resourceName := "aws_s3_bucket.bucket"
+
+	// record the initialized providers so that we can use them to check for the instances in each region
+	var providers []*schema.Provider
```
Comment (bflad) [inline aws/resource_aws_s3_bucket_test.go]: Thanks for adding this test! ❤️ 

Given this test is explicitly for testing within the same region, it seems like we should be able to nix the more complicated `ProviderFactories` setup for a normal test setup and inline both `aws_s3_bucket` resources into the same test configuration without provider aliases, etc. 😄

Comment (ewbankkit) [reply aws/resource_aws_s3_bucket_test.go]: Agreed. I'll simplify.

**Comment by ewbankkit** [comment]
OK, significantly simplified the test case and associated configuration(s).
Acceptance tests:
```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSS3Bucket_SameRegionReplicationSchemaV2'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSS3Bucket_SameRegionReplicationSchemaV2 -timeout 120m
=== RUN   TestAccAWSS3Bucket_SameRegionReplicationSchemaV2
=== PAUSE TestAccAWSS3Bucket_SameRegionReplicationSchemaV2
=== CONT  TestAccAWSS3Bucket_SameRegionReplicationSchemaV2
--- PASS: TestAccAWSS3Bucket_SameRegionReplicationSchemaV2 (61.66s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	61.749s
```

File: aws/resource_aws_s3_bucket_test.go [hunk 2]
```diff
@@ -3767,6 +3859,28 @@ resource "aws_s3_bucket" "destination" {
 `, randInt, randInt, randInt)
 }
 
+func testAccAWSS3BucketReplicationConfig_iamPolicy(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_iam_role" "test" {
+  name = %[1]q
```
Comment (ewbankkit) [inline aws/resource_aws_s3_bucket_test.go]: IAM role name matches [sweeper prefixes](https://github.com/terraform-providers/terraform-provider-aws/blob/8d319bdb8719ef1ff5de60fe9b230b93f3a1f777/aws/resource_aws_iam_role_test.go#L52-L60) by virtue of `tf-test-bucket-` prefix on bucket names.

**Comment by ewbankkit** [comment]
Rebased to remove merge conflicts.
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSS3Bucket_SameRegionReplicationSchemaV2'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSS3Bucket_SameRegionReplicationSchemaV2 -timeout 120m
=== RUN   TestAccAWSS3Bucket_SameRegionReplicationSchemaV2
=== PAUSE TestAccAWSS3Bucket_SameRegionReplicationSchemaV2
=== CONT  TestAccAWSS3Bucket_SameRegionReplicationSchemaV2
--- PASS: TestAccAWSS3Bucket_SameRegionReplicationSchemaV2 (50.53s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	50.585s
```

**Pull request merged**